src/viam/sdk/tests/test_resource.cpp

```diff
@@ -0,0 +1,172 @@
+#define BOOST_TEST_MODULE test module test_resource
+#include <boost/test/included/unit_test.hpp>
+
+#include <viam/sdk/common/pose.hpp>
+#include <viam/sdk/config/resource.hpp>
+#include <viam/sdk/referenceframe/frame.hpp>
+#include <viam/sdk/resource/resource_api.hpp>
+#include <viam/sdk/spatialmath/geometry.hpp>
+#include <viam/sdk/spatialmath/orientation.hpp>
+#include <viam/sdk/spatialmath/orientation_types.hpp>
+
+namespace viam {
+namespace sdktests {
+
+using namespace viam::sdk;
+
+BOOST_AUTO_TEST_CASE(test_api) {
+    API api1;
+    api1.set_namespace("ns");
+    BOOST_CHECK_EQUAL(api1.type_namespace(), "ns");
+    api1.set_resource_type("component");
+    BOOST_CHECK_EQUAL(api1.resource_type(), "component");
+    api1.set_resource_subtype("st");
+    BOOST_CHECK_EQUAL(api1.resource_subtype(), "st");
+    BOOST_CHECK_EQUAL(api1.to_string(), "ns:component:st");
+    BOOST_CHECK(!api1.is_service_type());
+    BOOST_CHECK(api1.is_component_type());
+
+    API api2 = API::from_string("ns:service:st");
+    BOOST_CHECK_EQUAL(api2.to_string(), "ns:service:st");
```

Comment:
I chose not to because I extensively checked the functions with `api1`.
The `to_string()` function check also technically checks if the namespace and subtype is correct.
Should I still add it? There isn't a reason not to.

Reply:
I think it's still useful. I don't think that we're showing here that `from_string` does the right thing w/r/t `namespace` and `subtype`, just `to_string`. It might be a bit overlapping but no harm in being explicit!